pkg/reconciler/instances/istio/actions/performer.go

```diff
@@ -75,52 +77,66 @@ type DataPlaneVersion struct {
 	IstioVersion string `json:"IstioVersion,omitempty"`
 }
 
+type CommanderResolver interface {
+	GetCommander(version istioctl.Version) (istioctl.Commander, error)
+}
+
 //go:generate mockery --name=IstioPerformer --outpkg=mock --case=underscore
 // IstioPerformer performs actions on Istio component on the cluster.
 type IstioPerformer interface {
 
-	// Install Istio on the cluster.
-	Install(kubeConfig, manifest string, logger *zap.SugaredLogger) error
+	// Install Istio on the cluster. Tries to find best matching istioctl binary for the 'requestedVersion'.
```

Comment:
Isn't this doc a little bit too much about implementation details? From my perspective this doc could better sound like:
`Intall Istio on the cluster with requestedVersion using istioChart`.
In that way you could also change the order of the parameters to be more readable, so first would go `requestedVersion` and later `istioChart`.

By the way, this function itself does not really provide an implementation of "finding the best matching istioctl binary" - it delegates the work to another struct. It should be documented there. In that way you will not repeat yourself in every function signature (DRY rule). Again, the fact that it uses `istioctl` under the hood is the implementation detail. If we really want to document it that way I would mention it in `DefaultIstioPerformer` implementation - in general you don't have to install Istio using the binary, right? The contract should be unchanged, the implementation details should be hidden, especially in the interface.

Reply:
:+1 I simplified the docs + mentioned `istioctl` usage  on the `DefaultIstioPerformer` docs.